### PR TITLE
Error checking, esp for Play.pulse

### PR DESCRIPTION
**Pull request by lcapelluto**
Raise an error if Play is called with a pulse that isn't a Pulse. Raise an error also if Instruction gets a negative duration. Cast ShiftPhase phase argument to float

### Summary

thank you, @eggerdj 

He pointed out that this:
```
sched = inst_map.get(args)
sched += Play(sched, chan)
```
should error but does not

### CODE REVIEW

**Comment by lcapelluto** [comment]
Cyclic import errors. This is because Pulse has to support `__call__` which returns `Play`, and `Play` checks that `pulse` is a `Pulse`.... not sure if I'll be successful but I'll try again later

File: qiskit/pulse/instructions/play.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
 from typing import Optional
 
 from ..channels import PulseChannel
+from ..exceptions import PulseError
+from ..pulse_lib.pulse import Pulse
```
Comment (SooluThomas) [inline qiskit/pulse/instructions/play.py:22]: `tox -elint` isn't showing any error locally. I'm guessing importing `Pulse` is causing the cyclic import. Can you please try removing `Pulse` imports and type hints?

**Comment by taalexander** [comment]
> Cyclic import errors. This is because Pulse has to support `__call__` which returns `Play`, and `Play` checks that `pulse` is a `Pulse`.... not sure if I'll be successful but I'll try again later

Is it possible to disable to the errors in this situation? If not one solution to this is to not import `Play` in `__call__` until within the call method. This will ensure that `Play` has been loaded when called for this first time. See the approach [here](https://github.com/Qiskit/qiskit-terra/blob/02621d2875346b4509aa87c31b5947932f5be7f8/qiskit/circuit/quantumcircuit.py#L1260) for an example.

**Comment by lcapelluto** [comment]
> > Cyclic import errors. This is because Pulse has to support `__call__` which returns `Play`, and `Play` checks that `pulse` is a `Pulse`.... not sure if I'll be successful but I'll try again later
> 
> Is it possible to disable to the errors in this situation? If not one solution to this is to not import `Play` in `__call__` until within the call method. This will ensure that `Play` has been loaded when called for this first time. See the approach [here](https://github.com/Qiskit/qiskit-terra/blob/02621d2875346b4509aa87c31b5947932f5be7f8/qiskit/circuit/quantumcircuit.py#L1260) for an example. I think this is an indication of poor design, but it is a solution.

@taalexander if you check out the code, you'll see I did this. I agree that it's an indicator of poor design -- luckily, an already deprecated design, so it won't be an issue in a few months.

**Comment by taalexander** [comment]
The design was my fault. Better late than never...

**Pull request merged**